pkg/apis/wmc/v1alpha1/windowsmachineconfig_types.go

```diff
@@ -41,11 +42,43 @@ type Azure struct {
 
 // WindowsMachineConfigStatus defines the observed state of WindowsMachineConfig
 type WindowsMachineConfigStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// ProvisionedVMCount is the number of running VMs provisioned by the operator
+	ProvisionedVMCount int `json:"createdVMCount"`
+	// JoinedVMCount is the number of VMs that successfully joined the cluster
+	JoinedVMCount int `json:"joinedVMCount"`
+	// Conditions states the latest available observations of current state
+	Conditions []WindowsMachineConfigCondition `json:"conditions"`
+}
+
+type WindowsMachineConfigCondition struct {
+	// Type describes the type of the condition
+	Type WindowsMachineConfigConditionType `json:"type"`
+	// Status of the condition
+	Status corev1.ConditionStatus `json:"status"`
+	// LastTransitionTime is the timestamp corresponding to the last status change of this condition
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// Reason is the machine readable reason for the condition change
+	Reason string `json:"reason"`
+	// Message is the human readable reason for the condition change
+	Message string `json:"message"`
 }
 
+type WindowsMachineConfigConditionType string
+
+const (
+	// CloudProviderAPIConnection indicates if we were able to use the cloud provider API with the provided credentials
+	CloudProviderAPIConnected WindowsMachineConfigConditionType = "CloudProviderAPIConnected"
+	// VMProvisionFailure indicates if there was an issue provisioning a new VM
+	VMProvisionFailure WindowsMachineConfigConditionType = "VMProvisionFailure"
+	// VMTerminationFailure indicates if there was an issue terminating an existing VM
+	VMTerminationFailure WindowsMachineConfigConditionType = "VMTerminationFailure"
+	// Reconciling indicates if the controller is currently attempting to reconcile
+	Reconciling WindowsMachineConfigConditionType = "Reconciling"
+)
+
```

Comment:
Should this be moved into `helpers.go/conditions.go`

Reply:
Yes i think thats a good change to make